### PR TITLE
Manchester | 26-ITP-Jan | Liban Jama | Sprint 3 | Practice tdd

### PR DESCRIPTION
**Pull request by libanj0161**
## Learners, PR Template

Self checklist

- [x] I have titled my PR with Region | Cohort | FirstName LastName | Sprint | Assignment Title
- [x] My changes meet the requirements of the task
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)

## Changelist

Complete practice tdd.

## Questions

No.

### CODE REVIEW

File: Sprint-3/2-practice-tdd/repeat-str.js [hunk 1]
```diff
@@ -1,5 +1,8 @@
-function repeatStr() {
-  return "hellohellohello";
+function repeatStr(str, count) {
+  if (count < 0) {
+    throw new Error("invalid count");
+  }
+  return str.repeat(count);
 }
```
Comment (Theoreoluwa) [inline Sprint-3/2-practice-tdd/repeat-str.js:27]: This function does not cover all the edge cases as expected to be implemented for the task. Then, your test would not run successfully because it's not working according to the task brief. Please, look at it again and read the comments

File: Sprint-3/2-practice-tdd/repeat-str.test.js [hunk 2]
```diff
@@ -21,12 +21,33 @@ test("should repeat the string count times", () => {
...
 // When the repeatStr function is called with these inputs,
 // Then it should return an empty string.
 
+test("should repeat the string count times", () => {
+  const str = "hello";
+  const count = 0;
+  const repeatedStr = repeatStr(str, count);
+  expect(repeatedStr).toEqual("");
+});
+
```
Comment (Theoreoluwa) [inline Sprint-3/2-practice-tdd/repeat-str.test.js:48]: let your function be implemented to suit these test cases

File: Sprint-3/2-practice-tdd/get-ordinal-number.js [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+  if (lastdigit === 2) {
+    return `${num}nd`;
+  }
+
+  if (lastdigit === 3) {
+    return `${num}rd`;
+  } else {
+    return `${num}th`;
+  }
 }
```
Comment (Theoreoluwa) [inline Sprint-3/2-practice-tdd/get-ordinal-number.js:31]: Your function needs to handle edge cases for other groups of numbers that depend on the last two digits as well. After this, your test should run successfully.

File: Sprint-3/2-practice-tdd/count.test.js [hunk 4]
```diff
@@ -17,8 +17,30 @@ test("should count multiple occurrences of a character", () => {
...
+  const count = countChar(str, char);
+  expect(count).toEqual(1);
+});
+
+test("should count multiple occurrences of a character", () => {
+  const str = "playful";
+  const char = "l";
+  const count = countChar(str, char);
+  expect(count).toEqual(2);
+});
```
Comment (Theoreoluwa) [inline Sprint-3/2-practice-tdd/count.test.js:32]: Be careful about using a duplicated test description when writing different tests.

**Review by Theoreoluwa — COMMENTED** [review]
You've done well, Liban. Look at the comments and make it better. Rooting for you.

File: Sprint-3/2-practice-tdd/count.test.js (hunk 4 above)
Comment (libanj0161) [reply Sprint-3/2-practice-tdd/count.test.js:32]: i've changed up the description so they are different.

File: Sprint-3/2-practice-tdd/get-ordinal-number.js (hunk 3 above)
Comment (libanj0161) [reply Sprint-3/2-practice-tdd/get-ordinal-number.js:31]: Ive changed the function and added tests to handle edge cases.

File: Sprint-3/2-practice-tdd/repeat-str.js (hunk 1 above)
Comment (libanj0161) [reply Sprint-3/2-practice-tdd/repeat-str.js:27]: I realised once reading the task brief and changed it to meet the requirements, thanks for the reminder.

File: Sprint-3/2-practice-tdd/repeat-str.test.js (hunk 2 above)
Comment (libanj0161) [reply Sprint-3/2-practice-tdd/repeat-str.test.js:48]: ive altered it, thanks.

File: Sprint-3/2-practice-tdd/repeat-str.js [hunk 5]
```diff
@@ -1,5 +1,29 @@
...
+    return str;
+  }
 
+  let result = "";
+
+  for (let i = 0; i < count; i++) {
+    result += str;
+  }
+
+  return result;
```
Comment (Theoreoluwa) [inline Sprint-3/2-practice-tdd/repeat-str.js:25]: I would approve this because I know I asked you to change it, but the way you handled the function in your previous pr looked cleaner and more readable and handled all test cases. Please stick to that as you move on.

**Review by Theoreoluwa — COMMENTED** [review]
Great job Liban, keep going!

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
